cm31_ntt/src/ntt.rs

```diff
@@ -0,0 +1,1233 @@
+use serde::{Serialize, Deserialize};
+use crate::cm31::CF;
+use num_traits::{Zero, One, Pow};
+use crate::ntt_utils::*;
+
+pub const NTT_BLOCK_SIZE_FOR_CACHE: usize = 8usize.pow(5);
+
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+pub struct PrecomputedTwiddles {
+    pub small: Vec<CF>,  // for the base NTT of size NTT_BLOCK_SIZE_FOR_CACHE
+    pub full: Vec<CF>,   // for the higher levels of recursion in the radix-8 context
+    pub stride: Vec<CF>, // for stride 2 or stride 4
+}
+
+pub fn precompute_twiddles(n: usize) -> Result<PrecomputedTwiddles, String> {
+    assert!(is_power_of(n as u32, 2), "n must be a power of 2");
```

Comment:
You can use `eyre::Result<PrecomputedTwiddles>` or `anyhow::Result<PrecomputedTwiddles>` instead of `Result<..., String>`. This will make it easier to nail down the errors as both these types can link to source.

With that, this assert can be replaced with something like `anyhow::ensure!(is_power_of(n as u32, 2), "n must be a power of 2");`